pyk/src/pyk/ktool/kprint.py

```diff
@@ -49,6 +58,31 @@ def __init__(self, kompiled_directory: str):
         self.symbol_table = build_symbol_table(self.definition, opinionated=True)
         self.definition_hash = hash_str(self.definition)
 
+    def kore_to_kast(self, kore: Kore) -> KAst:
+        with NamedTemporaryFile('w') as ntf:
+            ntf.write(kore.text)
+            ntf.flush()
+            _LOGGER.info(f'Wrote file: {ntf.name}')
+            kast_command = ['kast', '--input', 'kore', '--output', 'json', ntf.name, '--definition', str(self.kompiled_directory)]
+            command_env = os.environ.copy()
```

Comment:
You get an empty env if you don't specifically do this inheritance. This is how you pass the env along.

Reply:
Are you sure? If `PATH` wasn't passed along, that would make the default pretty useless. By not specify, I mean pass `None`, or don't provide and argument, not pass in the empty dict.

From `subprocess.run`'s documentation, it seems that the default is to inherit:
> If env is not None, it must be a mapping that defines the environment variables for the new process; these are used instead of the default behavior of inheriting the current process’ environment. It is passed directly to Popen.

Reply:
Huh, I thought we were discarding it because we had exactly the `PATH` problem before, so we started manually inheriting. But it appears that maybe this is a breaking in `run_process`, where it should default to inheriting instead of overriding: https://github.com/runtimeverification/k/blob/a3b3f34bb6d08179ee5bc7d3489d8f98b0f46bd6/pyk/src/pyk/cli_utils.py#L52.

Issue made: https://github.com/runtimeverification/k/issues/2769

This change (that the issue describes) will end up making a lot more changes than just here, so I'd rather it be a separate PR.

Reply:
Hmm, looking at the implementation of `run_process`, it seems to be doing the right thing. I'm surprised its getting an empty environment.

Reply:
No the default env is `None`: https://github.com/runtimeverification/k/blob/a3b3f34bb6d08179ee5bc7d3489d8f98b0f46bd6/pyk/src/pyk/cli_utils.py#L43

Reply:
Right, that invokes the default behaviour for  `subprocess.popen`.